src/main/java/net/minecraftforge/debug/item/ItemInCreativeTabTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.item;
+
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.item.ItemEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+@Mod(modid = ItemInCreativeTabTest.MODID, name = "Item In CreativeTab Test", version = "1.0", acceptableRemoteVersions = "*")
+public class ItemInCreativeTabTest {
+    public static final String MODID = "itemincreativetabtest";
+
+    private static final boolean ENABLED = false;
+    private static Logger logger;
+
+    @Mod.EventHandler
+    public void preInit(FMLPreInitializationEvent event) {
+        logger = event.getModLog();
+
+        if (ENABLED) {
+            MinecraftForge.EVENT_BUS.register(ItemInCreativeTabTest.class);
+        }
+    }
+
+    @SubscribeEvent
+    public static void onInCreativeTab(final ItemEvent.InCreativeTab event) {
+        if (event.getCreativeTab().equals(CreativeTabs.INVENTORY)) {
+//            logger.info("not making Item " + event.getItem().getUnlocalizedName() + " appear in CreativeTab " + event.getCreativeTab().getTabLabel() + " because the creative tab \"inventory\" is just the Destroy Item button & should not have anything displayed in it");
+            return;
+        }
+
+        if (event.getItem().getRegistryName().getResourcePath().contains("z")) {
```

Comment:
this should probably be changed to specific Items to make it clear which Items are affected (currently BlazeRod, BlazeDust and Quartz)